applications/FemToDemApplication/custom_processes/generate_dem_process.cpp

```diff
@@ -132,14 +132,24 @@ void GenerateDemProcess::CreateDEMParticle(
     rNode.SetValue(IS_DEM, true);
     rNode.SetValue(RADIUS, Radius);
     rNode.SetValue(DEM_PARTICLE_POINTER, spheric_particle);
+
+    // We transfer kinematic information
+    if (rNode.SolutionStepsDataHas(VELOCITY)) {
+        const array_1d<double, 3> vel = rNode.FastGetSolutionStepValue(VELOCITY);
```

Comment:
I think you don't need a copy here.

Reply:
Yeah I doubted about that... I want to apply to the particle the same value but afterwards should be free to be different. I suppose there is no problem with that

Reply:
Well, both options work, but the point is that you're allocating something only for the sake of assigning the value below. I'm fully convinced that you can achieve de same with a reference (I don't see your point about fixity).